python/lsst/pipe/tasks/computeExposureSummaryStats.py

```diff
@@ -331,33 +376,38 @@ def update_psf_stats(
         # 750bffe6620e565bda731add1509507f5c40c8bb/src/PsfFlux.cc#L112
         summary.psfArea = float(np.sum(im.array)/np.sum(im.array**2.))
 
-        if image_mask is not None:
-            psfApRadius = max(self.config.minPsfApRadiusPix, 3.0*summary.psfSigma)
-            self.log.debug("Using radius of %.3f (pixels) for psfApFluxDelta metric", psfApRadius)
-            psfTraceRadiusDelta, psfApFluxDelta = compute_psf_image_deltas(
-                image_mask,
-                psf,
-                sampling=self.config.psfGridSampling,
-                ap_radius_pix=psfApRadius,
-                bad_mask_bits=self.config.psfBadMaskPlanes
-            )
-            summary.psfTraceRadiusDelta = float(psfTraceRadiusDelta)
-            summary.psfApFluxDelta = float(psfApFluxDelta)
-            if image_ap_corr_map is not None:
-                if self.config.psfApCorrFieldName not in image_ap_corr_map.keys():
-                    self.log.warn(f"{self.config.psfApCorrFieldName} not found in "
-                                  "image_ap_corr_map.  Setting psfApCorrSigmaScaledDelta to NaN.")
-                    psfApCorrSigmaScaledDelta = nan
-                else:
-                    image_ap_corr_field = image_ap_corr_map[self.config.psfApCorrFieldName]
-                    psfApCorrSigmaScaledDelta = compute_ap_corr_sigma_scaled_delta(
-                        image_mask,
-                        image_ap_corr_field,
-                        summary.psfSigma,
-                        sampling=self.config.psfGridSampling,
-                        bad_mask_bits=self.config.psfBadMaskPlanes,
-                    )
-                summary.psfApCorrSigmaScaledDelta = float(psfApCorrSigmaScaledDelta)
+        if self.config.psfGridSampling is None:
+            self.log.info("Note: not computing grid-based PSF & ApCorr model fidelity metrics. Setting "
+                          "psfTraceRadiusDelta, psfApFluxDelta, & psfApCorrSigmaScaledDelta to NaN.")
+        else:
+            if image_mask is not None:
```

Comment:
Why is image_mask optional? I can't think of any case where we'd not have access to the mask, and certainly the three places that we call this method, we pass a mask.

Reply:
Please ask about that on slack and file a ticket if there's no known reason: as I said, all three places in the stack that currently call this pass `mask`, and it doesn't make sense for it to be optional to me.

Reply:
I'm gonna push back on this again.  It's pre-existing and unrelated to this ticket.  I am not bothered by it, so would not feel inclined to weigh in on, let alone initiate, a discussion about it.  If you feel strongly about it, proceed as you will!